src/nccl_ofi_rdma.cpp

```diff
@@ -5170,29 +5171,26 @@ static int listen_close(nccl_net_ofi_listen_comm_t *listen_comm)
 	rdma_endpoint_get_device((nccl_net_ofi_rdma_ep_t *)base_ep)->comm_idpool->free_id(l_comm->comm_id);
 
 	free(l_comm);
-	ret = base_ep->release_ep(base_ep, false, false);
+	ret = base_ep->release_ep(false, false);
 
 	return ret;
 }
 
-static int listen(nccl_net_ofi_ep_t *base_ep,
-			     nccl_net_ofi_conn_handle_t *handle,
-			     nccl_net_ofi_listen_comm_t **listen_comm)
+int nccl_net_ofi_rdma_ep_t::listen(nccl_net_ofi_conn_handle_t *handle,
+				   nccl_net_ofi_listen_comm_t **listen_comm)
 {
 	int ret = 0;
 	nccl_net_ofi_rdma_listen_comm_t *l_comm = NULL;
 	size_t comm_id = 0;
-	nccl_net_ofi_rdma_ep_t *ep =
-		(nccl_net_ofi_rdma_ep_t *)base_ep;
-	nccl_net_ofi_ep_rail_t *first_control_rail = rdma_endpoint_get_control_rail(ep, 0);
+	nccl_net_ofi_ep_rail_t *first_control_rail = rdma_endpoint_get_control_rail(this, 0);
 
 	/* Retrieve and validate device */
-	nccl_net_ofi_rdma_device_t *device = rdma_endpoint_get_device(ep);
+	nccl_net_ofi_rdma_device_t *device = rdma_endpoint_get_device(this);
 	assert(device != NULL);
 
 	int dev_id = device->base.dev_id;
 
-	ret = post_rx_buffs(ep);
+	ret = post_rx_buffs(this);
```

Comment:
more specific to the comment above.  If you're passing `this` as the first argument, it should just be a member function :).